contracts/KeeperRegistry.sol

```diff
@@ -379,15 +379,15 @@ contract KeeperRegistry is
     require(msg.sender == address(LINK), "only callable through LINK");
     require(data.length == 32, "data must be 32 bytes");
     uint256 id = abi.decode(data, (uint256));
-    validateUpkeep(id);
+    require(s_upkeep[id].maxValidBlocknumber > block.number.add(CANCELATION_DELAY), "upkeep must be active");
```

Comment:
Same suggestion. Doesn't seem like we should accept payment at all all after cancellation, only in-flight upkeeps.